test/namestate-test.js

```diff
@@ -0,0 +1,341 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+/* eslint no-unused-vars: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const NameState = require('../lib/covenants/namestate');
+const rules = require('../lib/covenants/rules');
+const Network = require('../lib/protocol/network');
+
+const network = Network.get('regtest');
+
+const {
+  treeInterval,
+  biddingPeriod,
+  revealPeriod,
+  transferLockup,
+  renewalWindow,
+  claimPeriod,
+  auctionMaturity,
+  lockupPeriod
+} = network.names;
+
+describe('Namestate', function() {
+  describe('open auction name', function() {
+    const name = 'handshake';
+    const nameHash = rules.hashName(name);
+    let height = 0;
```

Comment:
Can this be rewritten as set of unit tests instead of shared state between tests?

Reply:
Yeah you got me I'm sloppy. OK this has been updated in b6b55b28599a4392ac1aacae0cd7ef1cd3e836cf

I still used a single `describe` block to run through the single linear auction flow, but then after that the "forked timeline" stuff has been replaced with hard coded start heights for the block the auction ends on.